ui/webui/src/components/app.jsx

```diff
@@ -67,26 +73,32 @@ export const Application = () => {
 
             setAddress(address);
 
+            const errorContext = {
+                context: _("Initialize storage model."),
+                hint: _("This should not happen, the installation is doomed."),
+            };
+
+            const errorHandler = onCritFail(errorContext);
+
             Promise.all([
-                initDataStorage({ dispatch }),
+                initDataStorage({ dispatch, errorHandler }),
```

Comment:
I don't see the reason to pass the 'errorHandler' here. Any exception that happens inside this function, if the promises are correctly chained should be possible to catch in the end of the chain, with a catch or as a second argument to 'then', as it's already done is line 91.

Reply:
The point is to pass the handler to the getDeviceDataAction so that the exceptions caught here: https://github.com/rvykydal/anaconda/blob/a415a9f093d1661e6a9df8fe31f6699018584d84/ui/webui/src/actions/storage-actions.js#L61 are handled by it instead by console.error. (Or we need to remove the catch from the action).

Reply:
I think we should remove it then. If we agree that this action if failed is a critical error , it should never have the plain 'console.error' handler.

Reply:
Ah, it didn't work for me before https://github.com/rhinstaller/anaconda/commit/8ef0b9d508b120433cbc8c039424bbfda2cf0602